operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,490 @@
+
+SLS (System Layout Services) REST APIs are used to map these management nodes xnames to the corresponding `Kubernetes` and storage nodes names. This mapping of rack to `Kubernetes` and storage nodes is stored in the below format as a JSON file to be consumed by the `Kubernetes` and `Ceph` zoning modules later.
+
+**Example of node mapping JSON file**
+
+`rr_hw_discovery.json`
+
+```text
+{
+x3000:[ncn-m001,ncn-w001,ncn-w004,ncn-w007,ncn-s001],
+x3001:[ncn-m002,ncn-w002,ncn-w006,ncn-w005,ncn-w008],
+x3002:[ncn-m003,ncn-w003,ncn-w009,ncn-s005,ncn-s002]
+}
+```
+
+#### 1.2 Validate distribution of management/ managed nodes
+
+The discovery results (`rr_hw_discovery.json`) from step 1.1 above are passed to the placement validation module where it is validated whether the current placement meets the required criteria for enabling rack resiliency.
+
+![Rack Resiliency solution overview](../../img/rack-resiliency-2.png)
+
+The placement validation algorithm, as shown in the flow chart above decides whether the current placement of management nodes is suitable for enabling rack resiliency. If it is found that the current placement is not suitable for rack resiliency the next step which involves zoning for `Kubernetes` and `Ceph` is not performed.
+
+This module also evaluates if managed nodes are present in the management rack and informational messages are generated for the same.
+
+Note: Slingshot switch placement discovery and validation is out of scope for this release (CSM 1.7.0) of Rack Resiliency solution.
+
+### 2 Spreading of NCNs across `Kubernetes` zones, and spreading storage nodes across `Ceph` zones
+
+Upon successful validation of placement, using the node mapping `rr_hw_discovery.json` from discovery module, Zoning is performed
+
+#### 2.1 Zoning `Kubernetes` nodes 
+
+The `Kubernetes` topology zoning can be used to apply labels to the racks/ nodes in order to create management failure domains (MFD).
+
+The node is labeled with the key `topology.kubernetes.io/zone=<zone-id>`, where `<zone-id>` is of the form `Rack-1`, `Rack-2`, and so on. These labels can be used to identify all the management nodes which belong to the same zone and is used to schedule the critical services across the zones. 
+
+
+#### 2.2 Zoning `Ceph`
+
+`Ceph` provides the CRUSH map algorithm which can help segregate the storage nodes across zones. Using a combination of CRUSH rules and bucket types (hosts, racks, rows, etc.), the data can be replicated across zones. The objective of `Ceph` zoning is to make sure data gets replicated at rack level, so that there would not be data loss in case of a rack failure.
+
+#### 2.2.1 Creating `Ceph` zones with CRUSH 
+
+* Create racks by creating `Ceph` buckets and map storage nodes to these buckets using 
+    * "`Ceph` osd crush add-bucket <rack_name> rack"
+    * "`Ceph` osd crush move {bucket-name} {bucket-type}={bucket-name}, [...]"

```

Comment:
Isn't the command name for this all lowercase?

Reply:
These commands are fine, but yes they need to be updated:
```suggestion
    * `ceph osd crush add-bucket <rack_name> rack`
    * `ceph osd crush move {bucket-name} {bucket-type}={bucket-name}, [...]`

```